### PR TITLE
Bom Ver Search

### PR DESCRIPTION
**Pull request by RossStewart**

### CODE REVIEW

File: src/Persistence.LinnApps/Repositories/BomVerificationHistoryRepository.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        public BomVerificationHistoryRepository(ServiceDbContext serviceDbContext)
+            : base(serviceDbContext.BomVerificationHistory)
+        {
+            this.serviceDbContext = serviceDbContext;
+        }
+
+        public override IQueryable<BomVerificationHistory> FilterBy(Expression<Func<BomVerificationHistory, bool>> expression)
+        {
+            return this.serviceDbContext.BomVerificationHistory
+                .Include(x => x.PartNumber)
```
Comment (lewisrenfrew) [inline src/Persistence.LinnApps/Repositories/BomVerificationHistoryRepository.cs:25]: you shouldn't .Include() simple string fields like this

File: src/Service.Host/client/src/components/BomVerificationHistoryUtility/BomVerificationHistorySearch.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import Grid from '@mui/material/Grid';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import Stack from '@mui/material/Stack';
+import { useSelector, useDispatch } from 'react-redux';
+import { Page, collectionSelectorHelpers, Search } from '@linn-it/linn-form-components-library';
+import bomVerificationHistoryEntriesSearchActions from '../../actions/bomVerificationHistoryEntriesActions';
+import history from '../../history';
+import config from '../../config';
+import { bomVerificationHistoryEntries as bomVerificationHistoryEntriesItemType } from '../../itemTypes';
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/BomVerificationHistoryUtility/BomVerificationHistorySearch.js:11]: probably don't need the 'as bomVerificationHistoryEntriesItemType' alias here, can just import it as is

File: src/Service.Host/client/src/components/BomVerificationHistoryUtility/BomVerificationHistorySearch.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            'partNumber',
+            'remarks'
+        )
+    );
+    const searchLoading = useSelector(state =>
+        collectionSelectorHelpers.getSearchLoading(state.bomVerificationHistoryEntries)
+    );
+
+    const goToSelectedEntry = selectedEntry => {
+        history.push(`${bomVerificationHistoryEntriesItemType.uri}/${selectedEntry.tRef}`);
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/BomVerificationHistoryUtility/BomVerificationHistorySearch.js:35]: you can use the rel: self link on the resource here

File: src/Service.Host/client/src/components/BomVerificationHistoryUtility/BomVerificationHistorySearch.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    };
+
+    const goToCreate = () => {
+        history.push(`${bomVerificationHistoryEntriesItemType.uri}/create`);
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/BomVerificationHistoryUtility/BomVerificationHistorySearch.js:39]: could also possibly add a rel: create link to the resource for this one, for niceness

File: src/Service/Modules/BomVerificationHistoryModule.cs [hunk 5]
```diff
@@ -19,14 +19,9 @@ public class BomVerificationHistoryModule : ICarterModule
...
         {
-            app.MapGet("/purchasing/bom-verification/", this.GetApp);
             app.MapGet("/purchasing/bom-verification/{id:int}", this.GetBomVerificationHistoryEntry);
-            app.MapPost("/purchasing/bom-verification/", this.CreateBomVerificationHistoryEntry);
-        }
-
-        private async Task GetApp(HttpRequest req, HttpResponse res)
-        {
-            await res.Negotiate(new ViewResponse { ViewName = "Index.html" });
+            app.MapPost("/purchasing/bom-verification/create", this.CreateBomVerificationHistoryEntry);
```
Comment (lewisrenfrew) [inline src/Service/Modules/BomVerificationHistoryModule.cs:23]: Why add the /create here?

**Review by lewisrenfrew — APPROVED** [review]

File: src/Service/Modules/BomVerificationHistoryModule.cs (hunk 5 above)
Comment (RossStewart) [reply src/Service/Modules/BomVerificationHistoryModule.cs:23]: Was looking at similar parts of App and that seemed to be a kind of running theme. Happy to change it back though!

**Pull request merged**